osu.Game/Rulesets/Edit/Checks/CheckAudioQuality.cs

```diff
@@ -26,13 +26,13 @@ public class CheckAudioQuality : ICheck
             new IssueTemplateNoBitrate(this)
         };
 
-        public IEnumerable<Issue> Run(IBeatmap playableBeatmap, IWorkingBeatmap workingBeatmap)
+        public IEnumerable<Issue> Run(IBeatmap beatmap, BeatmapVerifierContext context)
```

Comment:
Is it feasible to also move the `IBeatmap` inside the context?

Reply:
Yes, but `IBeatmap` is used in almost every single check so it would make things quite verbose. The idea is that this encapsulation stores info that not all checks will need.

Reply:
by verbose, you mean accessing `context.Beatmap` instead of `beatmap`?

Reply:
I prefer `context.Beatmap` in any case. Less potential for breakage (this is a ruleset API after all), and keeps things in one place.

Reply:
Yeah, I'd prefer doing just `beatmap` to `context.Beatmap` as it's used a lot. I doubt we'd be likely to change `IBeatmap` as it's a pretty general interface, so the potential for breakage is already quite low, unless I'm misunderstanding.

Reply:
I can imagine a future in which we change the parameter type from `IBeatmap` to something like `EditorBeatmap`, for example. Such change wouldn't break if it was placed in the context.

Reply:
I can only imagine the reason for doing so would be giving checks additional information about the editor itself, in which case it would probably be better to create a subclass of `ICheck` e.g. `EditorCheck` which specifically deals with the `EditorBeatmap`.

Reply:
... Or adding common editor functions, like `GetClosestSnappedTime` which was moved there for `CheckUnsnappedObjects` to be able to use it.

`EditorBeatmap` was a bad example - it should never be exposed as is because it has mutation functions.

Reply:
I think I'm loosely for putting the beatmap inside the context because it feels like an arbitrary choice to leave it out. The pattern here is one used in many places in .NET via configuration objects, and in my experience I've never seen the "main" part separated out.

Reply:
Sure, we'll go with that then. I considered `BeatmapVerifierContext` more as something supplementing the beatmap instance (its name probably communicated that poorly), but after trying both approaches I don't really mind too much either way.